src/audio/eq_iir/eq_iir.c

```diff
@@ -339,38 +340,32 @@ static void eq_iir_free_delaylines(struct comp_data *cd)
 		iir[i].delay = NULL;
 }
 
-static int eq_iir_setup(struct comp_data *cd, int nch)
+static int eq_iir_init_coef(struct sof_eq_iir_config *config,
+			    struct iir_state_df2t *iir, int nch)
 {
-	struct iir_state_df2t *iir = cd->iir;
-	struct sof_eq_iir_config *config = cd->config;
 	struct sof_eq_iir_header_df2t *lookup[SOF_EQ_IIR_MAX_RESPONSES];
 	struct sof_eq_iir_header_df2t *eq;
-	int64_t *iir_delay;
-	int32_t *coef_data, *assign_response;
-	size_t s;
-	size_t size_sum = 0;
+	int32_t *assign_response;
+	int32_t *coef_data;
+	int size_sum = 0;
+	int resp = 0;
 	int i;
 	int j;
-	int resp;
-
-	/* Free existing IIR channels data if it was allocated */
-	eq_iir_free_delaylines(cd);
+	int s;
 
-	trace_eq("eq_iir_setup(), "
-		 "channels_in_config = %u, number_of_responses = %u",
-		 config->channels_in_config, config->number_of_responses);
+	trace("eq_iir_init_coef(), response assign for %u channels, "
+	      "%u responses", config->channels_in_config,
+	      config->number_of_responses);
 
 	/* Sanity checks */
 	if (nch > PLATFORM_MAX_CHANNELS ||
 	    config->channels_in_config > PLATFORM_MAX_CHANNELS ||
 	    !config->channels_in_config) {
-		trace_eq_error("eq_iir_setup() error: "
-			       "invalid nch or channels_in_config");
+		trace_er("eq_iir_init_coef(), error: invalid channels count");
```

Comment:
Because this one with the original `trace_eq_error` shouldn't make Checkpatch complain, and if it does you may safely ignore it.